src/cheri/xycheriot.adoc

```diff
@@ -0,0 +1,511 @@
+
+* <<section_priv_cheri,{cheri_priv_m_ext}>>,
+
+* <<section_zylevels1>>,
+
+* <<section_zyseal>>, and
+
+* <<section_rvy_sentry_jalr_ext>>.
+
+The present specification presumes the *absence* of all of:
+
+* H-, S-, or U-mode,
+
+* <<section_cheri_hybrid_ext,{cheri_default_ext_name}>>,
+
+* <<section_cheri_disable,{cheri_priv_m_reg_enable_ext}>>,
+
+* <<section_rvy_sentry_insn_ext>>, and
+
+* ...
+
+[NOTE]
+=====
+While {cheriot_ext_name} is nominally compatible with
+<<section_cheri_hybrid_ext,{cheri_default_ext_name}>> and
+<<section_cheri_disable,{cheri_priv_m_reg_enable_ext}>>,
+and particular instantiations may opt to permit disabling CHERI,
+we have not yet found compelling reason to formally specify this composition.
+=====
+
+[NOTE]
+=====
+While {cheriot_ext_name} is nominally compatible with
+<<section_rvy_sentry_insn_ext>>,
+the operating systems written for {cheriot_ext_name} have security models
+that presume the absence of ambient sealing.
+=====
+
+=== New Instructions
+
+==== Loads and Stores With `x0` Authority
+
+Instructions such as `{load_cap_name_lc} {cd}, n(x0)` would,
+in {cheri_base_ext_name},
+necessarily fault due to the clear {ctag} of the all-zeros `x0`.
+{cheriot_ext_name} defines these instructions
+to use <<mtidc>> as the authority instead.
+
+[NOTE]
+=====
+TODO, research required: we have never actually tested this; it is just
+conjecture that it works.  Maybe this should be some non-{cheriot_ext_name}
+extension.
+=====
+
+[NOTE]
+=====
+TODO, research required: if we made this depend on <<pcc>>.<<asr_perm>>, using
+<<mtidc>> for unprivileged operations and <<mscratch>> for privileged ones, we
+might be able to get away from the annoying, type-system-wrecking, IRQs-deferred
+CSR/GPR exchange dance at the top of the switcher's trap handler.
+=====
+
+==== Conditional Branches
+
+{cheri_base_ext_name} reserves uncompressed `beq` and `bne` instruction
+encodings in which `rs1≥rs2`.
+{cheriot_ext_name} defines such instructions with `rs2=x0` to be
+conditional branches on the {ctag} of `rs1` being set (`beq`) or clear (`bne`).
+
+[NOTE]
+=====
+TODO, research required: we have never actually tested this; it is just
+conjecture that it works.  Maybe this should be some non-{cheriot_ext_name}
+extension.
+=====
+
+[#section_cheriot_auicgp,reftext="{cheriot_auicgp_name}"]
+==== {cheriot_auicgp_name}
+
+{cheriot_ext_name} defines a new U-type instruction, {cheriot_auicgp_name},
+which adds its shifted (see <<section_cheriot_auipc_shift>>) U-immediate
+to the address of <<mtidc>> and stores the result in a target register,
+analogously to <<AUIPC_CHERI>> and <<pcc>>.
+
+==== {cheriot_setbounds_big_imm_name}
+
+[NOTE]
+=====
+TODO: setbounds immediate as an I-type instruction w/ 12-bit immediate rather
+than the base's R-type w/ (scaled) 5-bit immediate.
+=====
+
+==== {cheriot_setbounds_round_down_name}
+
+[NOTE]
+=====
+TODO: setbounds register that fixes base and rounds length down for
```

Comment:
Would also really like to see this one as a non-CHERIoT-specific extension in the future.